extensions/arc/src/models/resourceModel.ts

```diff
@@ -5,14 +5,18 @@
 
 import { ResourceInfo } from 'arc';
 import * as vscode from 'vscode';
-import { Registration } from './controllerModel';
+import { ControllerModel, Registration } from './controllerModel';
 
 export abstract class ResourceModel {
 
 	private readonly _onRegistrationUpdated = new vscode.EventEmitter<Registration>();
 	public onRegistrationUpdated = this._onRegistrationUpdated.event;
 
-	constructor(public info: ResourceInfo, private _registration: Registration) { }
+	constructor(private _controllerModel: ControllerModel, public info: ResourceInfo, private _registration: Registration) { }
+
+	public get controllerModel(): ControllerModel {
```

Comment:
Just make this public readonly if you want others to be able to use it

Reply:
Done. Are you trying to save the function call of getter?

Reply:
It's just unnecessary - no reason to take up the extra space if you don't actually need the getter to do additional logic.